pkgs/development/python-modules/temporalio/default.nix

```diff
@@ -0,0 +1,72 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  cargo,
+  fetchFromGitHub,
+  pythonOlder,
+  poetry-core,
+  protobuf,
+  python-dateutil,
+  rustc,
+  rustPlatform,
+  setuptools,
+  setuptools-rust,
+  types-protobuf,
+  typing-extensions,
+}:
+
+buildPythonPackage rec {
+  pname = "temporalio";
+  version = "1.8.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "temporalio";
+    repo = "sdk-python";
+    rev = "1.8.0";
+    hash = "sha256-xJIb+U3hV6DDvfbfY20U/4rE2z0XzaESkehtcvG6m50=";
+    fetchSubmodules = true;
+  };
+
+  cargoDeps = rustPlatform.importCargoLock { lockFile = ./Cargo.lock; };
```

Comment:
Why is cargo lock used instead of `fetchCargoTarball`?

Reply:
I figured out what was missing for this, thank you for the review.